subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -308,41 +309,54 @@ static void nrf_cloud_decode_desired_obj(cJSON *root_obj,
 	}
 }
 
-int nrf_cloud_encode_shadow_data(const struct nrf_cloud_sensor_data *sensor,
-				 struct nrf_cloud_data *output)
+#if defined(CONFIG_NRF_CLOUD_ALERTS)
```

Comment:
is the #if defined necessary?

Reply:
Well, I made the feature configurable in Kconfig, so if it is disabled, then this function won't be called.  It should compile though.  I suppose I could remove the #if defined if you feel strongly about it.

Reply:
OK, removing most of the CONFIG_NRF_CLOUD_ALERTS checks, and moved it inside the codec function.  I'll remove the conditional in CMakeLists.txt so it will still build an empty function.